inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/dialect/PostgresDialect.java

```diff
@@ -137,4 +146,71 @@ public List<LogicalTypeRoot> unsupportedTypes() {
                 LogicalTypeRoot.SYMBOL,
                 LogicalTypeRoot.UNRESOLVED);
     }
+
+    @Override
+    public List<String> getAndSetPkNamesFromDb(String tableIdentifier, JdbcOptions jdbcOptions) {
+        PreparedStatement st = null;
+        try {
+            JdbcOptions jdbcExecOptions =
+                    JdbcOptions.builder()
+                            .setDBUrl(jdbcOptions.getDbURL() + "/"
+                                    + JdbcMultiBatchingComm.getTDbNameFromIdentifier(tableIdentifier))
+                            .setTableName(JdbcMultiBatchingComm.getTbNameFromIdentifier(tableIdentifier))
+                            .setDialect(jdbcOptions.getDialect())
+                            .setParallelism(jdbcOptions.getParallelism())
+                            .setConnectionCheckTimeoutSeconds(jdbcOptions.getConnectionCheckTimeoutSeconds())
+                            .setDriverName(jdbcOptions.getDriverName())
+                            .setUsername(jdbcOptions.getUsername().orElse(""))
+                            .setPassword(jdbcOptions.getPassword().orElse(""))
+                            .build();
+            SimpleJdbcConnectionProvider tableConnectionProvider = new SimpleJdbcConnectionProvider(jdbcExecOptions);
+            Connection conn = tableConnectionProvider.getOrEstablishConnection();
```

Comment:
Does this connection need to close?

Reply:
close when task exit,  implement in JdbcMultiBatchingOutputFormat.close